### PR TITLE
fix: adding gke node ranges to os-updates firewall policy

### LINKED ISSUE #648: GKE nodes cannot be provisioned because trafic blocked by malicious ip
when GKE nodes are deployed, they are trying to communicate with cloud.google.com. This dns translates into an ip address that is included in the malicious ip list thus blocking that egress traffic.

### PR DESCRIPTION
**Pull request by davelanglois-ssc**
closes #648 

when GKE nodes are deployed, they are trying to communicate with cloud.google.com. This dns translates into an ip address that is included in the malicious ip list thus blocking that egress traffic.

This fix includes : 
- the GKE CIDR block to the firewall policy allowing os-updates
- removal of the RFC5735 ip range from the different resources
- including GKE node, services and pods ranges into the network isolation policy

### CODE REVIEW

**Review by fmichaelobrien — DISMISSED** [review]
/lgtm

**Review by alaincormier-ssc — APPROVED** [review]

File: solutions/client-landing-zone/setters.yaml [hunk 1]
```diff
@@ -135,6 +135,7 @@ data:
     - "debian.org"
     - "deb.debian.org"
     - "ubuntu.com"
+    - "cloud.google.com"
```
Comment (alaincormier-ssc) [inline solutions/client-landing-zone/setters.yaml:138]: might be worth adding a comment above this setter for this particular entry?

File: solutions/client-landing-zone/README.md [hunk 2]
```diff
@@ -12,29 +12,29 @@ Package to create a client's folder hierarchy, logging resources and a network h
...
+| dns-nameservers                      | ["ns-cloud-a1.googledomains.com.", "ns-cloud-a2.googledomains.com.", "ns-cloud-a3.googledomains.com.", "ns-cloud-a4.googledomains.com."]                                                                                                               | array |     1 |
+| dns-project-id                       | dns-project-12345                                                                                                                                                                                                                                      | str   |     2 |
+| firewall-internal-ip-ranges          | [10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16, 240.0.0.0/4]                                                                                                                                                                                               | array |     5 |
```
[the remark this answers is not in the document]
Comment (davelanglois-ssc) [reply solutions/client-landing-zone/README.md:26]: 240.0.0.0/4 is no longer included in the ip plan so I will make the correction in the packages to reflect this

File: solutions/client-landing-zone/setters.yaml (hunk 1 above)
Comment (cartyc) [reply solutions/client-landing-zone/setters.yaml:138]: +1

File: solutions/client-landing-zone/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davelanglois-ssc) [reply solutions/client-landing-zone/README.md]: latest commit includes that change

File: solutions/client-landing-zone/setters.yaml (hunk 1 above)
Comment (davelanglois-ssc) [reply solutions/client-landing-zone/setters.yaml:144]: included in latest commit

**Review by alaincormier-ssc — DISMISSED** [review]

File: solutions/client-landing-zone/setters.yaml [hunk 3]
```diff
@@ -90,17 +90,23 @@ data:
...
   # Subnet IP range for nonp-main in northamerica-northeast1
   standard-nane1-nonp-main-snet: 10.1.0.0/21
   #
   # Subnet IP range for nonp-main in northamerica-northeast2
   standard-nane2-nonp-main-snet: 10.1.8.0/21
   #
-  # pbmm IP space
-  standard-pbmm-cidr: 10.1.128.0/18
+  # pbmm IP space, used by ComputeFirewallPolicyRule for network isolation
+  # this list must include all pbmm cidr range including the ranges for GKE nodes, services and pods
```
Comment (lucstjean-ssc) [inline solutions/client-landing-zone/setters.yaml:106]: Maybe add an s to the first range - range > ranges.

File: solutions/client-landing-zone/setters.yaml [hunk 4]
```diff
@@ -90,17 +90,23 @@ data:
   project-allowed-restrict-vpc-peering: |
     - under:projects/PROJECT_ID
   #
-  # nonp IP space
-  standard-nonp-cidr: 10.1.0.0/18
+  # nonp IP space, used by ComputeFirewallPolicyRule for network isolation
+  # this list must include all non protected cidr range including the ranges for GKE nodes, services and pods
```
Comment (lucstjean-ssc) [inline solutions/client-landing-zone/setters.yaml:94]: Maybe add an s to the first range - range > ranges.

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

File: solutions/client-landing-zone/setters.yaml (hunk 3 above)
Comment (davelanglois-ssc) [reply solutions/client-landing-zone/setters.yaml]: done in latest commit

File: solutions/client-landing-zone/setters.yaml (hunk 4 above)
Comment (davelanglois-ssc) [reply solutions/client-landing-zone/setters.yaml]: done in latest commit

**Review by alaincormier-ssc — DISMISSED** [review]

**Review by cartyc — DISMISSED** [review]
/lgtm Nice work Dave!

**Review by lucstjean-ssc — DISMISSED** [review]
/lgtm

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

**Review by lucstjean-ssc — APPROVED** [review]
/lgtm

**Review by fmichaelobrien — APPROVED** [review]
/lgtm

**Review by cartyc — APPROVED** [review]
/lgtm

**Pull request merged**